bug: `tpch` queries suddenly using `string_view`

## Related
- [x] https://github.com/pola-rs/polars/issues/26435

## Description
Starting from (https://github.com/pola-rs/polars/releases/tag/py-1.38.0) the queries are [failing](https://github.com/narwhals-dev/narwhals/actions/runs/21670750471/job/62477688240?pr=3449).

<details><summary><b>Show very long logs</b></summary>
<p>

```py

=========================== short test summary info ============================
FAILED tests/queries_test.py::test_execute_query[q11-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q15-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q22-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'utf8_slice_codeunits' has no kernel matching input types (string_view)
FAILED tests/queries_test.py::test_execute_query[q20-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, uint32)
FAILED tests/queries_test.py::test_execute_query[q17-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q11-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q13-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q7-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q19-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q6-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q4-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q4-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q20-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q2-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q12-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q19-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q8-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q22-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q9-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field p_name
FAILED tests/queries_test.py::test_execute_query[q20-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q9-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q2-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q1-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, uint32)
FAILED tests/queries_test.py::test_execute_query[q14-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q3-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'equal' has no kernel matching input types (string_view, string)
FAILED tests/queries_test.py::test_execute_query[q5-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field r_name
FAILED tests/queries_test.py::test_execute_query[q7-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'equal' has no kernel matching input types (string_view, string)
FAILED tests/queries_test.py::test_execute_query[q17-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q19-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field p_name
FAILED tests/queries_test.py::test_execute_query[q11-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field ps_comment
FAILED tests/queries_test.py::test_execute_query[q13-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'match_substring_regex' has no kernel matching input types (string_view)
FAILED tests/queries_test.py::test_execute_query[q15-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, uint32)
FAILED tests/queries_test.py::test_execute_query[q1-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int32)
FAILED tests/queries_test.py::test_execute_query[q17-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'equal' has no kernel matching input types (string_view, string)
FAILED tests/queries_test.py::test_execute_query[q18-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field o_orderstatus
FAILED tests/queries_test.py::test_execute_query[q5-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q21-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q16-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'match_substring_regex' has no kernel matching input types (string_view)
FAILED tests/queries_test.py::test_execute_query[q18-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q13-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q14-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field l_returnflag
FAILED tests/queries_test.py::test_execute_query[q12-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field o_orderstatus
FAILED tests/queries_test.py::test_execute_query[q10-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field c_name
FAILED tests/queries_test.py::test_execute_query[q10-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q21-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q8-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field p_name
FAILED tests/queries_test.py::test_execute_query[q6-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, uint32)
FAILED tests/queries_test.py::test_execute_query[q4-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field l_returnflag
FAILED tests/queries_test.py::test_execute_query[q2-pyarrow] - pyarrow.lib.ArrowInvalid: Data type string_view is not supported in join non-key field p_name
FAILED tests/queries_test.py::test_execute_query[q22-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q6-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int32)
FAILED tests/queries_test.py::test_execute_query[q1-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q16-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q3-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q5-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q7-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q9-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q12-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q10-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q21-pyarrow] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, uint32)
FAILED tests/queries_test.py::test_execute_query[q16-pandas[pyarrow]] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q3-dask] - NotImplementedError: string_view
FAILED tests/queries_test.py::test_execute_query[q14-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q8-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q18-pandas[pyarrow]] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int64)
FAILED tests/queries_test.py::test_execute_query[q15-dask] - pyarrow.lib.ArrowNotImplementedError: Function 'array_take' has no kernel matching input types (string_view, int32)
======================== 66 failed, 66 passed in 48.62s ========================
```

</p>
</details> 



I found (https://github.com/pola-rs/polars/pull/26236), which removed an experimental feature that looks like the cause for a schema change.

I don't believe I opted-into anything here, so I'm pretty surprised by the breakage:

https://github.com/narwhals-dev/narwhals/blob/204652c9467b4e1d549c837e5e8c5d73a395e786/tpch/generate_data.py#L174

https://github.com/narwhals-dev/narwhals/blob/204652c9467b4e1d549c837e5e8c5d73a395e786/tpch/generate_data.py#L186

The problem doesn't impact `polars` at all, as it doesn't expose a `DataType` (in python) like [`pyarrow.string_view`](https://arrow.apache.org/docs/python/generated/pyarrow.string_view.html).

IIUC, the TPCH queries are hit by this because of this conversion change:

- [`duckdb.sqltypes.VARCHAR`](https://duckdb.org/docs/stable/sql/data_types/text) -> [`ArrowDataType::Utf8`](https://docs.rs/polars-arrow/0.52.0/polars_arrow/datatypes/enum.ArrowDataType.html#variant.Utf8)
- `duckdb.sqltypes.VARCHAR` -> [`ArrowDataType::Utf8View`](https://docs.rs/polars-arrow/0.52.0/polars_arrow/datatypes/enum.ArrowDataType.html#variant.Utf8View)

And so `pyarrow`, `pandas[pyarrow]`, `dask` are now all broken because they have `*View` types.
SQL-like backends still work, as they natively understand `VARCHAR` as a `String`

--- a/.github/workflows/check_tpch_queries.yml
+++ b/.github/workflows/check_tpch_queries.yml
@@ -26,9 +26,7 @@ jobs:
           cache-suffix: ${{ matrix.python-version }}
           cache-dependency-glob: "pyproject.toml"
       - name: local-install
-        run: |
-          uv pip install -U -e ".[dask]" --group core-tests
-          uv pip install -U duckdb
+        run: uv pip install -U -e ".[dask]" --group core-tests
       - name: generate-data
         run: cd tpch && python generate_data.py --debug
       - name: tpch-tests 

